drivers/pinctrl/pinctrl_ti_k3.c

```diff
@@ -27,10 +35,35 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 	const struct device *dev = DEVICE_DT_GET(PINCTRL_NODE);
 	uintptr_t virt_reg_base = DEVICE_MMIO_GET(dev);
 
+	#ifdef CONFIG_SOC_AM62x_M4
```

Comment:
If this unlocking just for pinctrl?
If it is needed for other peripherals in the future, maybe create a reusable function in soc.h

Reply:
It is only required once.

Reply:
Move it to a `SYS_INIT()` call if it is only required once.
Currently it may be called multiple times

If it is not related to pinctrl and only needs calling once, then it could be done in a `SYS_INIT()` in soc.c

Reply:
Have added it to `SYS_INIT()`